File: ssTools/kukijson/java/src/Kukijson.java
import com.google.gson.Gson;
import com.google.gson.JsonSyntaxException;
import com.google.gson.annotations.SerializedName;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

class Data {
    @SerializedName("nama")
    String[] nama;

    @SerializedName("isi")
    String[] isi;
}

public class Kukijson {
    public static void main(String[] args) {
        if (args.length != 2 || !args[0].equals("-f")) {
            System.out.println("Penggunaan: java Main -f <nama_file.json>");
            System.exit(1);
        }

        String filename = args[1];
        String jsonData = readJsonFile(filename);

        Gson gson = new Gson();
        Data data;
        try {
            data = gson.fromJson(jsonData, Data.class);
        } catch (JsonSyntaxException e) {
            System.out.println("Gagal memparse JSON");
            System.exit(1);
            return;
        }

        String formattedData = formatData(data);
        System.out.println(formattedData);
    }

    private static String readJsonFile(String filename) {
        StringBuilder sb = new StringBuilder();
        try (BufferedReader br = new BufferedReader(new FileReader(filename))) {
            String line;
            while ((line = br.readLine()) != null) {
                sb.append(line);
            }
        } catch (IOException e) {
            System.out.println("Gagal membuka file: " + filename);
            System.exit(1);
        }
        return sb.toString();
    }

    private static String formatData(Data data) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < data.nama.length; i++) {
            sb.append(data.nama[i]).append("=").append(data.isi[i]);
            if (i != data.nama.length - 1) {
                sb.append(";");
            }
        }
        return sb.toString();
    }
}

File: ssTools/kukijson/go/kukijson.go
package main

import (
	"encoding/json"
	"flag"
	"fmt"
	"io/ioutil"
	"os"
)

type Data struct {
	Nama []string `json:"nama"`
	Isi  []string `json:"isi"`
}

func main() {
	filename := flag.String("f", "", "Nama file JSON")
	flag.Parse()

	if *filename == "" {
		fmt.Println("Penggunaan: go run kukijson.go -f <nama_file.json>")
		os.Exit(1)
	}

	jsonData, err := ioutil.ReadFile(*filename)
	if err != nil {
		fmt.Printf("Gagal membuka file: %s\n", *filename)
		os.Exit(1)
	}

	var data Data
	err = json.Unmarshal(jsonData, &data)
	if err != nil {
		fmt.Println("Gagal memparse JSON")
		os.Exit(1)
	}

	formattedData := formatData(data)
	fmt.Println(formattedData)
}

func formatData(data Data) string {
	result := ""

	for i := 0; i < len(data.Nama); i++ {
		nama := data.Nama[i]
		isi := data.Isi[i]

		result += fmt.Sprintf("%s=%s", nama, isi)

		if i != len(data.Nama)-1 {
			result += ";"
		}
	}

	return result
}
